x-pack/plugins/secops/public/components/timeline/index.tsx

```diff
@@ -4,82 +4,143 @@
+  width: number;
+}
+
+interface StateProps {
   dataProviders: DataProvider[];
-  height?: string;
-  onColumnSorted: OnColumnSorted;
-  onDataProviderRemoved: OnDataProviderRemoved;
-  onFilterChange: OnFilterChange;
-  onRangeSelected: OnRangeSelected;
-  rowRenderers: RowRenderer[];
+  data: ECS[];
+  range: Range;
   sort: Sort;
-  width: number;
 }
 
-const TimelineDiv = styled.div<{ width: string; height: string }>`
-  display: flex;
-  flex-direction: column;
-  min-height: 700px;
-  overflow: none;
-  user-select: none;
-  width: ${props => props.width};
-  height: ${props => props.height};
-`;
-
-const defaultHeight = '100%';
-
-/** The parent Timeline component */
-export const Timeline = pure<Props>(
-  ({
-    columnHeaders,
-    columnRenderers,
-    dataProviders,
-    data,
-    height = defaultHeight,
-    onColumnSorted,
-    onDataProviderRemoved,
-    onFilterChange,
-    onRangeSelected,
-    rowRenderers,
-    sort,
-    width,
-  }) => (
-    <TimelineDiv data-test-subj="timeline" width={`${width}px`} height={height}>
-      <TimelineHeader
-        dataProviders={dataProviders}
-        onDataProviderRemoved={onDataProviderRemoved}
-        width={width}
-      />
-      <Body
-        columnHeaders={columnHeaders}
+interface DispatchProps {
+  createTimeline: ActionCreator<{ id: string }>;
+  addProvider: ActionCreator<{
+    id: string;
+    provider: DataProvider;
+  }>;
+  updateData: ActionCreator<{
+    id: string;
+    data: ECS[];
+  }>;
+  updateProviders: ActionCreator<{
+    id: string;
+    providers: DataProvider[];
+  }>;
+  updateRange: ActionCreator<{
+    id: string;
+    range: Range;
+  }>;
+  updateSort: ActionCreator<{
+    id: string;
+    sort: Sort;
+  }>;
+  removeProvider: ActionCreator<{
+    id: string;
+    providerId: string;
+  }>;
+}
+
+type Props = OwnProps & StateProps & DispatchProps;
+
+class StatefulTimelineComponent extends React.PureComponent<Props> {
+  public componentDidMount() {
+    const { createTimeline, id } = this.props;
+
+    createTimeline({ id });
+  }
+
+  public render() {
+    const {
+      data,
+      dataProviders,
+      headers,
+      id,
+      range,
+      removeProvider,
+      sort,
+      updateRange,
+      updateSort,
+      width,
+    } = this.props;
+
+    const onColumnSorted: OnColumnSorted = sorted => {
+      updateSort({ id, sort: sorted });
+    };
+
+    const onDataProviderRemoved: OnDataProviderRemoved = dataProvider => {
+      removeProvider({ id, providerId: dataProvider.id });
+    };
+
+    const onRangeSelected: OnRangeSelected = selectedRange => {
+      updateRange({ id, range: selectedRange });
+    };
+
+    return (
+      <Timeline
+        columnHeaders={headers}
         columnRenderers={columnRenderers}
         dataProviders={dataProviders}
         data={data}
         onColumnSorted={onColumnSorted}
         onDataProviderRemoved={onDataProviderRemoved}
-        onFilterChange={onFilterChange}
+        onFilterChange={noop}
```

Comment:
I take it you want to pass down a `noop` here? A comment beside it why maybe?

Reply:
Yes, that is the intent. That's the callback for column filters, which are out of scope for this phase. I'll add a comment per your suggestion, thanks.